File: ResourceManager/Configuration/HostFactory.cs
using Microsoft.Extensions.Hosting;

namespace ResourceManager.Configuration
{
    internal static class HostFactory
    {
        public static IHost Create()
        {
            var hostBuilder = Host.CreateDefaultBuilder()
                .ConfigureAppConfiguration(Startup.ConfigureAppConfiguration)
                .ConfigureServices(Startup.ConfigureServices)
                .ConfigureLogging(Startup.ConfigureLogging);

            return hostBuilder.Build();
        }
    }
}

File: ResourceManager/Configuration/HostFactory.vb
Imports Microsoft.Extensions.Hosting

Namespace ResourceManager.Configuration
    Friend Module HostFactory
        Public Function Create() As IHost
            Dim hostBuilder = Host.CreateDefaultBuilder().ConfigureAppConfiguration(AddressOf ConfigureAppConfiguration).ConfigureServices(AddressOf ConfigureServices).ConfigureLogging(New Action(Of HostBuilderContext, Microsoft.Extensions.Logging.ILoggingBuilder)(AddressOf ConfigureLogging))

            Return hostBuilder.Build()
        End Function
    End Module
End Namespace
